src/components/BreadCrumb/BreadCrumb.tsx

```diff
@@ -0,0 +1,66 @@
+import { Breadcrumb, BreadcrumbItem, PageBreadcrumb } from '@patternfly/react-core';
+import { isUndefined } from 'lodash';
+import { useMatches } from 'react-router-dom';
+
+import { TBreadCrumbAttributes, breadCrumbAttributes } from 'common/breadCrumbAttributes';
+
+import { uiLogger } from 'services/uiLogger';
+
+export interface ISingleBreadCrumb {
+  entity: string;
+  data: string | undefined;
+}
+
+export interface IBreadCrumbProps {
+  pageBreadCrumbs: ISingleBreadCrumb[];
+}
+
+export const BreadCrumb = ({ pageBreadCrumbs }: IBreadCrumbProps) => {
+  const matches = useMatches();
+  const matchCrumbs = matches.filter((match) => !isUndefined(match.handle));
+  const pageCrumbs = pageBreadCrumbs as ISingleBreadCrumb[];
+
+  const breadCrumbs = matchCrumbs.map((crumb, index) => {
+    const notLast = index < matchCrumbs.length - 1;
+    if (crumb.handle && typeof crumb.handle === 'string') {
+      const crumbHandle: TBreadCrumbAttributes = crumb.handle as TBreadCrumbAttributes;
+
+      let text;
+      if (pageCrumbs) {
+        const filteredCrumb = pageCrumbs.filter((pCrumb) => pCrumb.entity === crumbHandle);
+        if (filteredCrumb.length === 0) {
+          text = breadCrumbAttributes[crumbHandle].title;
+        } else if (filteredCrumb.length === 1) {
+          text = filteredCrumb[0].data
+            ? filteredCrumb[0].data
+            : breadCrumbAttributes[filteredCrumb[0].entity as TBreadCrumbAttributes].title;
+        } else {
+          uiLogger.error('More crumbs of same type comming from pages.');
+        }
+      }
+      if (notLast) {
+        return (
+          <BreadcrumbItem key={crumb.id} to={crumb.pathname}>
+            {text}
+          </BreadcrumbItem>
+        );
+      }
```

Comment:
```
const breadCrumbTitle = pageCrumbs?.find((pageCrumb) => pageCrumb.entity === crumbHandle);

return (
  <BreadcrumbItem ...>
     ...
    {breadCrumbTitle?.data ? readCrumbTitle.data : breadCrumbAttributes[crumbHandle].title}
  </BreadcrumbItem>
);
```

i would omit uiLogger call here

Reply:
We reworked it a bit but you're right that title is displayed until data is loaded. For some cases it is default behavior and in others it still provides basic navigation with working links even when data is not loaded for some reason

Reply:
In the future we agreed to integrate serviceContainer functionality. Possibly full service container will be added or some other simplified solution will be done.